server/sonar-server/src/test/java/org/sonar/server/ce/ws/TaskActionTest.java

```diff
@@ -134,6 +127,74 @@ public void task_is_archived() throws Exception {
   }
 
   @Test
+  public void return_stacktrace_of_failed_activity_with_stacktrace_when_additionalField_is_set() {
+    userSession.login("john").setGlobalPermissions(SYSTEM_ADMIN);
+
+    CeActivityDto activityDto = createActivityDto("TASK_1")
+      .setErrorMessage("error msg")
+      .setErrorStacktrace("error stack");
+    setHasErrorStacktrace(activityDto, true);
+    persist(activityDto);
+
+    TestResponse wsResponse = ws.newRequest()
+      .setMediaType(PROTOBUF)
+      .setParam("id", "TASK_1")
+      .setParam("additionalFields", "stacktrace")
+      .execute();
+
+    WsCe.TaskResponse taskResponse = Protobuf.read(wsResponse.getInputStream(), WsCe.TaskResponse.PARSER);
+    WsCe.Task task = taskResponse.getTask();
+    assertThat(task.getId()).isEqualTo("TASK_1");
+    assertThat(task.getErrorMessage()).isEqualTo(activityDto.getErrorMessage());
+    assertThat(task.hasHasErrorStacktrace()).isTrue();
+    assertThat(task.getHasErrorStacktrace()).isTrue();
+  }
+
+  @Test
+  public void do_not_return_stacktrace_of_failed_activity_with_stacktrace_when_additionalField_is_not_set() {
+    userSession.login("john").setGlobalPermissions(SYSTEM_ADMIN);
+
+    CeActivityDto activityDto = createActivityDto("TASK_1")
+      .setErrorMessage("error msg")
+      .setErrorStacktrace("error stack");
+    setHasErrorStacktrace(activityDto, true);
+    persist(activityDto);
+
+    TestResponse wsResponse = ws.newRequest()
+      .setMediaType(PROTOBUF)
+      .setParam("id", "TASK_1")
+      .execute();
+
+    WsCe.TaskResponse taskResponse = Protobuf.read(wsResponse.getInputStream(), WsCe.TaskResponse.PARSER);
+    WsCe.Task task = taskResponse.getTask();
+    assertThat(task.getId()).isEqualTo("TASK_1");
+    assertThat(task.getErrorMessage()).isEqualTo(activityDto.getErrorMessage());
+    assertThat(task.hasHasErrorStacktrace()).isTrue();
+    assertThat(task.getHasErrorStacktrace()).isTrue();
```

Comment:
this check is useless, and it should be replaced by checking that the stacktrace is not returned